muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/forum/ForumController.java

```diff
@@ -220,10 +221,17 @@ public ForumThread createForumThread(/** @PermitContext **/ ForumArea forumArea,
   @Permit (ForumResourcePermissionCollection.FORUM_DELETEMESSAGES)
   public void deleteThread(@PermitContext ForumThread thread) {
     List<ForumThreadReply> replies = forumThreadReplyDAO.listByForumThread(thread);
+    Iterator<ForumThreadReply> replyIterator = replies.iterator();
+    while(replyIterator.hasNext()){
+      ForumThreadReply reply = replyIterator.next();
+      if(reply.getParentReply() != null){
+        forumThreadReplyDAO.delete(reply);
+        replyIterator.remove();
+      }
+    }
```

Comment:
Both ForumThreadReply and ForumThread have archived flags, so they should be used instead of hard delete.